crates/essential-rest-client/src/builder_client.rs

```diff
@@ -79,6 +79,25 @@ impl EssentialBuilderClient {
         Ok(output)
     }
 
+    /// Register program
+    ///
+    /// Creates a solution to the program registry predicate and submits it.
+    pub async fn register_program(
+        &self,
+        big_bang: &BigBang,
+        program: &Program,
+    ) -> anyhow::Result<ContentAddress> {
+        let program_ca = essential_hash::content_addr(program);
+        print_registering(&program_ca);
+        let program_solution =
+            register_program_solution(big_bang.program_registry.clone(), program);
+        let solution_set = SolutionSet {
+            solutions: vec![program_solution],
+        };
+        print_registered();
```

Comment:
While we've constructed the solution set here and are ready to send it, it might be misleading to print "**Registered** successfully" here as it's not really registered until a block is produced that includes the solution.

Perhaps this should be something like "**Submitted** program {ca} for registration", and after the `self.submit_solution_set`?

Reply:
Also "Submitted program {ca} for registration" should be printed after the next line where the solution set is submitted

Reply:
The print in the solution submission does not provide information about the content address of the contract / program that is being registered though

Reply:
Ah, indeed. 
I will make modifications to correct this.